src/PFire.Core/Protocol/Messages/Inbound/GameInformation.cs

```diff
@@ -12,5 +15,13 @@ internal sealed class GameInformation : XFireMessage
 
         [XMessageField("gport")]
         public int GamePort { get; set; }
+
+        public override async Task Process(IXFireClient context)
+        {
+            context.User.GameID = GameId;
+            context.User.GameIP = GameIP;
+            context.User.GamePort = GamePort;
+            await context.UpdateGameInfo();
```

Comment:
Can you move your `UpdateGameInfo` method here instead of calling from the client? It's not the best pattern, but it's consistent with how the other outbound message work.

Let's also call it `SendGameInfoToFriends(user)` where the user is `context.User`